src/integration-test/utils.ts

```diff
@@ -2,136 +2,100 @@ import { Observable } from '@reactivex/rxjs';
 import * as Boom from 'boom';
 import * as chalk from 'chalk';
 import { spawn } from 'child_process';
-import { merge } from 'lodash';
+import * as _debug from 'debug';
+import * as fs from 'fs';
+import * as path from 'path';
 
-import originalFetch, { RequestInit, Response as OriginalResponse } from 'node-fetch';
+const debug = _debug('system-integration-tests');
+import fetch, { Response } from 'node-fetch';
 import { ENV } from '../shared/types';
-import { Auth0, Config } from './types';
+import CharlesClient, { ResponseMulti, ResponseSingle } from './charles-client';
+import { Auth0, CharlesClients, CharlesResponse, Config, TeamType } from './types';
+
+const mkpath = require('mkpath');
 
 export function sleep(ms = 0) {
   return new Promise(resolve => setTimeout(resolve, ms));
 }
 
-export interface Response extends OriginalResponse {
-  tryJson: <T>(onlyOnSuccess?: boolean) => T;
+export function wrapResponse<T>(response: Response): CharlesResponse<T> {
+  const _response = response as any;
+  _response.toJson = getResponseJson<T>(response);
+  _response.getEntity = getEntity(_response);
+  _response.getEntities = getEntities(_response);
+  return _response;
 }
 
-export type Fetch = (url: string, options?: RequestInit) => Promise<Response>;
-
-export function fetchFactory(accessToken: string, retryCount = 0, sleepFor = 2000) {
+function getEntity(response: CharlesResponse<ResponseSingle>) {
+  return () => response.toJson().then(x => x.data);
+}
 
-  const innerFetch = async (url: string, options?: RequestInit) => {
-    const _options: RequestInit = merge({
-      redirect: 'manual',
-      headers: {
-        authorization: `Bearer ${accessToken}`,
-        cookie: `token=${accessToken}`,
-      },
-    }, options || {});
-    // These are here intentionally for debugging purposes
-    // console.log('--> HTTP %s %s', (_options && _options.method) || 'GET', url);
-    // console.dir(_options, { colors: true });
-    return wrapResponse(await originalFetch(url, _options));
-  };
-  let out = innerFetch;
-  if (retryCount > 0) {
-    out = async (url: string, options?: RequestInit) => {
-      for (let i = 0; i < retryCount; i++) {
-        try {
-          return await innerFetch(url, options);
-        } catch (err) {
-          log(`WARN: Fetch failed for url ${url}. Error message is '${err.message}'`);
-          await sleep(sleepFor);
-        }
-      }
-      throw Error(`Fetch failed ${retryCount} times for url ${url}`);
-    };
-  }
-  return out;
+function getEntities(response: CharlesResponse<ResponseMulti>) {
+  return () => response.toJson().then(x => x.data);
 }
 
-function wrapResponse(response: OriginalResponse): Response {
-  const _response = response as any;
-  _response.tryJson = async <T>(onlyOnSuccess = true) => {
-    if (!onlyOnSuccess || (response.status >= 200 && response.status < 300)) {
+export function getResponseJson<T>(response: Response) {
```

Comment:
Three, although none of them are major:

1. It's a bit more complex.
2. Unless there's a use case that I don't see, is there ever a need to call this twice for the same response? If not, this could hide a bug in the testing code.
3. It leaks memory since a closure with the parsed response is created each time the function is called.